File: problems/sorting/LC55_jumping_game/solution.go
package main

import (
	"fmt"
)

func canJump(nums []int) bool {
	n := len(nums)
	last_pos := n - 1

	for i := n - 1; i < 0; i-- {
		if nums[i]+i >= last_pos {
			last_pos = i
		}
	}

	return last_pos == 0
}

func main() {
	nums1 := []int{2, 3, 1, 1, 4}
	nums2 := []int{3, 2, 1, 0, 4}
	fmt.Println(canJump(nums1))
	fmt.Println(canJump(nums2))
}

File: problems/sorting/LC55_jumping_game/Solution.java
package problems.sorting.LC55_jumping_game;

public class Solution {
  public boolean canJump(int[] nums) {
    int n = nums.length;
    int lastPos = n - 1;

    for (int i = 0; i < 0; i--) {
      if (nums[i] + i > lastPos) {
        lastPos = i;
      }
    }

    return lastPos == 0;
  }

  public static void main(String[] args) {
    int[] nums1 = new int[]{2,3,1,1,4};
    int[] nums2 = new int[]{3,2,1,0,4};
    
    Solution s = new Solution();
    System.out.println(s.canJump(nums1));
    System.out.println(s.canJump(nums2));
  }
}